### DISCUSSION TITLE
참조 지역성

### DISCUSSION THREAD

**Discussion by 3Juhwan** [2024-woowacourse-study/level-interview#265] [Computer Architecture]
.

**Comment by le2sky** [comment]
참조 지역성의 원리(locality of reference)는 CPU가 메모리에 접근할 때 주된 경향을 바탕으로 만들어진 원리다. 캐시 메모리의 적중률을 높여 CPU의 메모리  접근 횟수를 줄이는데 이용된다. 

- **시간 지역성(temporal locality)** : CPU는 최근에 접근했던 메모리 공간에 다시 접근하려는 경향이 있다. ex) 프로그래밍에서 지역 변수에 값을 저장하면 이는 나중에 다시 접근하게 된다.
- **공간 지역성(spatial locality)** : CPU는 접근한 메모리 공간 근처를 접근하려는 경향이 있다. ex) CPU가 인텔리제이 프로그램을 실행할 때는 인텔리제이 프로그램이 모여 있는 공간 근처를 집중적으로 접근한다.

**Reply by le2sky** [reply]
이런곳에서도 쓰이구나 용으로 메모

[고성능을 위한 MySQL Primary Key 설계 전략](https://px201226.github.io/mysql-primary-key-design/)
[Tim sort에 대해 알아보자](https://d2.naver.com/helloworld/0315536)
